### ISSUE TITLE
Aspire doesn't generate Projects class correctly

### ISSUE THREAD

**Issue by macsux** [dotnet/aspire#2519]
I've installed Aspire as part of CLI installation method as described [here](https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/setup-tooling?tabs=dotnet-cli#install-net-aspire). Then created a new project based on `aspire-starter` template using CLI. When trying to build the project, I'm getting the following error

```
dotnet run --project AspireSample/AspireSample.AppHost
Building...
C:\projects\AspireSample\AspireSample.AppHost\Program.cs(5,46): error CS0234: The type or namespace name 'AspireSample_ApiService' does not exist in the namespace 'Projects' (are you missing an assembly reference?)
[C:\projects\AspireSample\AspireSample.AppHost\AspireSample.AppHost.csproj]
C:\projects\AspireSample\AspireSample.AppHost\Program.cs(7,29): error CS0234: The type or namespace name 'AspireSample_Web' does not exist in the namespace 'Projects' (are you missing an assembly reference?) [C:\pro
jects\AspireSample\AspireSample.AppHost\AspireSample.AppHost.csproj]

The build failed. Fix the build errors and run again.
```

It seems that `Projects` namespace should be getting autogenerated and include classes for every project in solution, yet it only contains a single class for the host project. The other ones are missing. This is what the generated file in `AspireSample.AppHost/obj/Debug/net8.0/Aspire/references/_AppHost.ProjectMetadata.g.cs` looks like:

```csharp
namespace Projects;

internal static class AspireSample_AppHost
{
  public static string ProjectPath => """C:\projects\AspireSample\AspireSample.AppHost""";
}
```

Here's info on my environment:
```
dotnet --list-sdks
3.1.426 [C:\Program Files\dotnet\sdk]
5.0.214 [C:\Program Files\dotnet\sdk]
6.0.419 [C:\Program Files\dotnet\sdk]
7.0.406 [C:\Program Files\dotnet\sdk]
8.0.100 [C:\Program Files\dotnet\sdk]
8.0.201 [C:\Program Files\dotnet\sdk]
```

```
dotnet workload list

Installed Workload Id      Manifest Version                      Installation Source
------------------------------------------------------------------------------------
aspire                     8.0.0-preview.3.24105.21/8.0.100      SDK 8.0.200
wasm-tools                 8.0.2/8.0.100                         VS 17.8.34330.188
```

**Comment by davidfowl** [comment]
What's the version of the package in your project?

**Comment by macsux** [comment]
```    <PackageReference Include="Aspire.Hosting" Version="8.0.0-preview.3.24105.21" />```

**Comment by eerhardt** [comment]
@macsux - can you attach a binlog by doing `dotnet run AspireSample/AspireSample.AppHost /bl` and grabbing the msbuild.binlog file that is outputted?

**Comment by macsux** [comment]
Here you go [msbuild.zip](https://github.com/dotnet/aspire/files/14440142/msbuild.zip)

**Comment by eerhardt** [comment]
For some reason the Aspire.Hosting.Sdk targets aren't getting loaded. From your binlog:

![image](https://github.com/dotnet/aspire/assets/8291187/b5425876-9afd-4217-a392-ad998ba684a2)

From the binlog of a successful build:
![image](https://github.com/dotnet/aspire/assets/8291187/dc055b91-f87e-4993-a744-a8d7b296adc0)

@joperezr - have you ever seen this situation before? It looks like the workload is installed.

**Comment by joperezr** [comment]
Yeah, I downloaded the binlog and I think I found the issue. Take a look at this particular line in the Imports section of the evaluation of the AppHost project:

![image](https://github.com/dotnet/aspire/assets/13854455/30bca40e-535f-4a4f-9168-472b3cc3ec7f)

If you see there, looks like it is not importing the workload targets due to the property `MSBuildEnableWorkloadResolver` being set to false, which disables using workload-related targets. I confirmed this by going all the way up in the binlog and looking at the environment, and it looks like you have that set as your environment variable. Un-setting that, or switching that to true, should make the issue go away.
